rust/gui-client/src-tauri/src/client/gui.rs

```diff
@@ -573,6 +575,10 @@ impl Controller {
                         .context("Couldn't hide Welcome window")?;
                 }
             }
+            Req::SystemTrayMenu(TrayMenuEvent::AddFavorite(resource_id)) => {
+                self.advanced_settings.favorite_resources.insert(resource_id);
+                self.refresh_favorite_resources().await?;
```

Comment:
I guess it is normal for the system tray to close once you apply an action? It is a bit tedious to use.

Reply:
Yeah. We might replace it with a custom Tauri window soon since Tauri's tray menu is missing a lot of features we have on macOS / phones

And I'm still secretly thinking, Linux users would be happy with an `nmcli` / `pacmd` equiv or a `top` like TUI, and with this new abstraction layer it would be easy to write.... too much fun though